convert2rhel/repo.py

```diff
@@ -47,3 +54,46 @@ def get_rhel_repoids():
     loggerinst.info("RHEL repository IDs to enable: %s" % ", ".join(repos_needed))
 
     return repos_needed
+
+
+def download_repofile(repofile_url):
+    """Download the official repofile pointing to a convert2rhel repository.
+
+    :raises exceptions.CriticalError: When the repository file can't be accessed.
+    :returns str: Path to a successfully downloaded repofile
```

Comment:
```suggestion
    """Download a repository file from a specific URL.

    :raises exceptions.CriticalError: When the repository file URL is inaccessible.
    :returns str: Path to a repofile if successfully downloaded. None if there was a problem with storing the downloaded repofile.
```

@r0x0d, I think that the function should raise an exception in both cases:
- when unable to access the URL
- when unable to store the downloaded content

Right now the former leads to raising an exception and the latter to returning None. That makes checking the result of the function difficult.

Reply:
We are not raising exceptions in both cases: 
* when unable to access the URL (URLError)
* when the contents of the response is empty

Since I split the code and removed the operation of writing from this function, I thought that we should check for empty response instead. We could go further and check for status code different from 200... Do you think that would be necessary?